src/storage.js

```diff
@@ -46,23 +56,104 @@ class AWSStorage {
       .createHash('sha256')
       .update(mount.url)
       .digest('hex');
-    this._bucket = `h3${sha256.substr(0, 60)}`;
+
+    this._bucket = `h3${sha256.substr(0, 59)}`;
+    this._region = region;
+    this._mountUrl = mount.url;
     this._log = log;
   }
 
-  async store(path, res) {
+  async _bucketExists() {
+    try {
+      await this.client.send(new HeadBucketCommand({
+        Bucket: this._bucket,
+      }));
+      return true;
+    } catch (e) {
+      /* istanbul ignore next */
+      if (e.$metadata.httpStatusCode !== 404) {
+        throw e;
+      }
```

Comment:
Head command either return success or throws a 404, everything else is exceptional